clients/fides-js/src/components/tcf/TcfOverlay.tsx

```diff
@@ -2,47 +2,110 @@ import { h, FunctionComponent } from "preact";
 import { useState, useCallback, useMemo } from "preact/hooks";
 import ConsentBanner from "../ConsentBanner";
 
-import { debugLog, hasActionNeededNotices } from "../../lib/consent-utils";
+import {
+  debugLog,
+  hasActionNeededNotices,
+  transformUserPreferenceToBoolean,
+} from "../../lib/consent-utils";
 
 import "../fides.css";
 import Overlay from "../Overlay";
-import { TcfConsentButtons } from "../ConsentButtons";
+import { TcfConsentButtons } from "./TcfConsentButtons";
 import { OverlayProps } from "../types";
 import TcfTabs from "./TcfTabs";
+import {
+  TCFFeatureRecord,
+  TCFPurposeRecord,
+  TCFVendorRecord,
+} from "../../lib/tcf/types";
+
+const resolveConsentValueFromTcfModel = (
+  model: TCFPurposeRecord | TCFFeatureRecord | TCFVendorRecord
+) => {
+  if (model.current_preference) {
```

Comment:
we don't handle notice-only yet—I originally didn't think there were "notice-only" TCF purposes, but I think there might be?

Reply:
I'm unsure. @rsilvery - are you aware of any TCF purposes being notice-only?